utilitiesformtg.php

```diff
@@ -1,352 +1,346 @@
-    } else {
-        return "Card not found";
-    }
-}
-
-/**
-* This function is made to extract 15 cards from a list of cards in a CSV file online
-* It needs a url to a CSV file published online and the column number where the cardnames are located
-*
-* [p1p1cube column="2"]https://example.com/mycube.csv[/p1p1cube]
-*/
-function ufmtg_randomize_card_pack($atts, $content = null) {
-    $spreadsheet_url = $content;
-    $column = (int)$atts['column'] -1; //minus 1 is to make it more intuitive for users, colum A=1, B=2 etc.
-    $cube = array();
-
-    if(!ini_set('default_socket_timeout', 15)) {
-        echo "unable to change socket timeout";
-    } 
-
-    if (($handle = fopen($spreadsheet_url, "r")) !== FALSE) {
-        while (($data = fgetcsv($handle, 1000, ",")) !== FALSE) {
-            $spreadsheet_data[] = $data;
-        }
-        fclose($handle);
-    } else {
-        die("Problem reading csv");
-    }
-
-    //Get the cards from the right column of the spreadsheet
-    foreach ($spreadsheet_data as $card1) {
-        $cube[] = $card1[$column];
-    }
-
-    //Shuffle them up so they aren't sorted
-    $random_pack = array_rand($cube, 15);
-    shuffle($random_pack);
-
-    $cardpack = "";
-    foreach($random_pack as $cardnum) {
-        $cardpack = $cardpack . $cube[$cardnum] . ":";
-    }
-    $cardpack = substr($cardpack, 0, -1);
-    
-    return ufmtg_get_list_from_scryfall($cardpack, true);
-}
-
-/**
- * [mtglink]
- */
-function ufmtg_get_mtg_link($atts, $content = null) {
-    $cardName = $atts['name'];
-    $cardLink = $atts['url'];
-    $cardBackSide = $atts['backside'] ?: "";
-    if ($cardBackSide !== "") {        
-        return '<span class="scryfall_hover_img"><a href="' . $cardLink . '" target="_blank">' . $cardName . '<span><img src="' . $content . '" class="scryfall_card-hover-size"/><img src="' . $cardBackSide . '" class="scryfall_card-hover-size"/></span></a></span>';
-    } else {
-        return '<span class="scryfall_hover_img"><a href="' . $cardLink . '" target="_blank">' . $cardName . '<span><img src="' . $content . '" class="scryfall_card-hover-size"/></span></a></span>';
-    }
-}
-
-/**
- * [mtgimg]
- */
-function ufmtg_get_mtg_image($atts, $content = null) {
-    $cardBackSide = $atts['backside'] ?: "";
-    return ufmtg_build_img("", $content, $cardBackSide);
-}
-
-function ufmtg_precache_cards($atts, $content = null) {
-    extractCardsFromString($content, "[scrylink]", "[/scrylink]");
-}
-
-add_shortcode('scryimg', 'ufmtg_get_scryfall_image');
-add_shortcode('scrylink', 'ufmtg_get_scryfall_link');
-add_shortcode('p1p1cube', 'ufmtg_randomize_card_pack');
-add_shortcode('p1p1deck', 'ufmtg_build_deck');
-add_shortcode('mtgimg', 'ufmtg_get_mtg_image');
-add_shortcode('mtglink', 'ufmtg_get_mtg_link');
-add_shortcode('mtgprecache', 'ufmtg_precache_cards');
-
-?>
```

Comment:
closing the php tag is not needed and is considered dangerous (if you add some whitespace below `?>` the output might start to early, breaking things that want to set headers)